sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -0,0 +1,301 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from contextlib import contextmanager
+from typing import Dict, Type, Callable, List
+import uuid
+import logging
+import time
+import threading
+from enum import Enum
+
+from azure.core.tracing import SpanKind  # type: ignore
+from azure.core.settings import settings  # type: ignore
+
+from azure.eventhub import EventPosition, EventHubError, EventData
+from .partition_context import PartitionContext
+from .partition_manager import PartitionManager, OwnershipLostError
+from .ownership_manager import OwnershipManager
+
+log = logging.getLogger(__name__)
+
+
+class CloseReason(Enum):
+    SHUTDOWN = 0  # user call EventProcessor.stop()
+    OWNERSHIP_LOST = 1  # lose the ownership of a partition.
+
+
+class EventProcessor(object):  # pylint:disable=too-many-instance-attributes
+    """
+    An EventProcessor constantly receives events from one or multiple partitions of the Event Hub
+    in the context of a given consumer group.
+
+    """
+    def __init__(
+            self, eventhub_client, consumer_group_name: str,
+            event_handler: Callable[[PartitionContext, List[EventData]], None],
+            *,
+            partition_id: str = None,
+            partition_manager: PartitionManager = None,
+            initial_event_position=EventPosition("-1"), polling_interval: float = 10.0,
```

Comment:
This makes it 3.x only, shouldn't this file be 2.7 compatible?

Reply:
Yes, this should be changed because it's sync and should support 2.7. They're copied from the async version. We'll change it. Good catch